src/views/Pathways/Courses/form/course.js

```diff
@@ -0,0 +1,42 @@
+import * as yup from "yup"
+
+const createdAtField = {
+  name: "createdAt",
+  type: "date",
+  validation: yup.date().typeError("required"),
+  customProps: {
+    label: "Created at",
+    variant: "dialog",
+    disabled: true,
+  },
+}
+
+export const getCourseAddFormStructure = (course,allCourses) => {
+
+  const optionsList=[]
+  let optionsToChoose= allCourses ? allCourses.map(eachcourse => optionsList.push({value: eachcourse.name, name:eachcourse.id })) : []
```

Comment:
it should just be `let options = allCourses ? allCourses.forEach(eachcourse => optionsList.push({value: eachcourse.name, name:eachcourse.id })) : []`

map is only supposed to be used when you are outputting something. if you just need to loop, use forEach.